gigascience/gigadb-website
# Spreadsheet tool deployment pipeline

# Pull request for issues: #1068, #1079, #1022

This is a pull request for the following functionalities:

* Build and deploy the dataset spreadsheet tool from Gitlab pipeline
* Disable yii2-queue-monitor functionality as its migrations are disfunctional
* Draft SOP for the tool
* Update to the dataset spreadsheet master template to fix typo with dataset types
* Staging build and deployment jobs now happen automatically

## How to test?

* Clone my branch
* Provision a new staging environment from it
* Run a complete Gitlab pipeline by pushing my branch to your remote fork of gigadb-website and wait for the triggered pipeline to finish)
* Follow the Dataset Upload SOP's instructions.

>Note: You can use the dataset spreadsheet for DOI 102240 sent by Chris A. to us as the candidate dataset spreadsheet to use when applying the SOP. It was sent to us by email with subject "Re: Create dataset 102240 (***Test***) (GIGA-D-22-00026) (DeePVP)" (ensure you use the v3 version attached to that email, earlier versions will fail)

## How have functionalities been implemented?

* A new build job and a new deploy job were created for the ``excel-spreadsheet-uploader`` GigaDB tool.
* The ``bastion_playbook.yml``  was updated to only deploy configuration and a wrapper shell script for the dataset upload workflow.
* To enable production container, I've created dedicated Dockerfiles and docker-compose files: ``gigadb/app/tools/excel-spreadsheet-uploader/Dockerfile-Production`` and ``gigadb/app/tools/excel-spreadsheet-uploader/docker-compose.production.build.yml``
* The ``execute.sh`` wrapper shell script was updated so it can run in-container locally or on host when on bastion.
* The ``execute.sh`` wrapper shell script can run on staging and live cloud deployments by relying on GIGADB_ENV being set in ``/home/centos/.bash_profile`` which is done as a task in ``bastion_playbook.yml``


## Any issues with implementation?

n/a

## Any changes to automated tests?

Updated ``protected/tests/functional/AdminDatasetMoveFilesActionTest.php``
to make the test ``testMoveFiles()`` more verbose in case of failure.

## Any changes to documentation?

Added SOP for dataset upload

## Any technical debt repayment?

n/a

## Any improvements to CI/CD pipeline?

Implement #1022 to make staging build and deploy jobs automated instead of manual
